### PR TITLE
Makefile modification to compile with no warning and only the necessary files

### PR DESCRIPTION
**Pull request by grypoB**
Reviewed the current Makefile and modified how the compilation was done to remove the following issues:
- The warnings due to the usage of wrong flags for compiling C++ (instead of C)
- Not all files would be recompiled each time but only the ones with their dependencies changed

For this, a new set of flags for C++ compilation was created, and the way ``headercheck.py`` (for RBSM communication) and ``fingerprint.py`` (for compilation time) are run during compilation was modified.

``main.cpp`` was changed to clear some warning due to the presence of the new c++ flag ``-Wmissing-declarations`` (equivalent of the c ``-Wmissing-prototypes``).

### CODE REVIEW

**Comment by ramensandwich** [comment]
Has this been tested and its output runs fine on the buggy?

File: Legacy/real_time/arduino_src/radio_buggy_mega/Makefile [hunk 1]
```diff
@@ -467,51 +489,46 @@ extcoff: $(TARGET).elf
 
 
 # Link: create ELF output file from object files.
-.SECONDARY : $(TARGET).elf
```
Comment (ramensandwich) [inline Legacy/real_time/arduino_src/radio_buggy_mega/Makefile]: Do we not care about keeping the ELF files anymore? If not, also delete the comment

Comment (grypoB) [inline Legacy/real_time/arduino_src/radio_buggy_mega/Makefile]: I'm not sure what this line does, I have been trying to compile with and without and haven't seen a difference

**Comment by grypoB** [comment]
This branch has been merged into buckley-cpp-changes (see #463)

**Pull request closed without merging**